ement-room.el

```diff
@@ -2817,151 +2809,113 @@ MSG is the message STRING was contained in."
-BUF is the room buffer this link is from. NUM is the index for this link."
-  (cond ((alist-get 'errcode data) (message "Ement: You are being rate limited."))
-        (data
-         (with-current-buffer buf
-           (let* ((id (ement-event-id msg))
-                  (related-node
-                   (ement-room--ewoc-node-before ement-ewoc
-                                                 msg
-                                                 (lambda (a b)
-                                                   (string= (ement-event-id a)
-                                                            (ement-event-id b)))
-                                                 :pred #'ement-event-p))
-                  (key (intern (concat "preview-" (number-to-string num)))))
-             (unless (alist-get key (map-elt (ement-event-local related)
-                                             (intern (concat "url-previews-" id))))
-               (cl-pushnew (cons key data) (map-elt (ement-event-local related)
-                                                    (intern (concat "url-previews-" id)))))
-             (unless (> num (or (car-safe (map-elt (map-elt (ement-event-local related)
-                                                            (intern (concat "url-previews-" id)))
-                                                   'url-preview-n))
-                                most-positive-fixnum))
-               (cl-pushnew `(url-preview-n ,num)
-                           (map-elt (ement-event-local related)
-                                    (intern (concat "url-previews-" id)))))
-             (ewoc-invalidate ement-ewoc related-node))))))
+NUM is the index for this link."
+  (cond ((alist-get 'errcode data) (progn
+                                     (message "Ement: You are being rate limited. Waiting 10s before trying again.")
+                                     (setq-local ement-room--preview-stop t)
+                                     (let ((saved-room (current-buffer)))
+                                       (run-with-timer 10 nil
+                                                       (lambda ()
+                                                         (with-current-buffer saved-room
+                                                           (setq-local ement-room--preview-stop nil)))))))
+        (data (let ((related-node (ement-room--ewoc-node-before ement-ewoc
+                                                                msg
+                                                                (lambda (a b)
+                                                                  (string= (ement-event-id a)
+                                                                           (ement-event-id b)))
+                                                                :pred #'ement-event-p))
+                    (key (intern (concat "preview-" (number-to-string num)))))
```

Comment:
Interning shouldn't be necessary for this.  If you need to do an alist lookup by string, do, e.g. `(alist-get KEY LIST nil nil #'equal)`.

Reply:
I don't think `equal` works for string-symbol comparisons? Only string-string or symbol-symbol I think.

I could use strings instead of symbols for all the preview keys but that feels...weird and unsettling :D I could do it that way if you wish though.

Reply:
Correct, objects of different type are not `equal`.

I don't think we should intern more symbols to the obarray here if we can avoid it.  We should be able to key on the URL, I think.